package.json

```diff
@@ -0,0 +1,65 @@
+{
+  "name": "maker-docker-poa",
+  "version": "0.0.1",
+  "description": "Maker SAI compiler and deployer",
+  "author": "Keydonix <team@keydonix.com>",
+  "repository": "http://github.com/keydonix/maker-docker-poa",
+  "main": "output/index.js",
+  "types": "output/index.d.ts",
+  "module": "output/index.js",
+  "browser": "output/browser.js",
+  "license": "GPL-3.0",
+  "directories": {
+    "src": "source"
+  },
+  "scripts": {
+    "build:source": "npx tsc",
+    "build:contracts": "npm run fetch:sai && node output/deployment/compileContracts.js",
+    "build:interfaces": "npm run fetch:sai && node output/tools/generateContractInterfaces.js",
+    "build": "npm run build:source && npm run build:contracts && npm run build:interfaces",
+    "clean": "rimraf output",
+    "clean-start": "npm run clean && npm run build",
+    "docker:build": "docker build . -t keydonix/maker-docker-poa",
+    "preversion": "npm run build",
+    "version": "git add source/libraries/ContractInterfaces.ts",
+    "release": "npm version prerelease && npm publish",
+    "fetch:sai": "bash fetch-sai-contracts.sh"
```

Comment:
I have never personally been a fan of all of these scripts.  In this case, since this code is expressly intended to be used inside a docker image, it feels like we should at least delete the ones that reference `docker`.  Personally, I would recommend clearing out all of the others that aren't required for `npm run clean-start`.

In fact, if it were just me I would delete them all and just put the full command to build directly into the Dockerfile so we can remove one layer of indirection since there is only every one command and only ever one place that command will be called (from the Dockerimage `RUN` line).

I think the command we want is:
```
npx tsc \
    && ./fetch-sai-contracts.sh \
    && node output/deployment/compileContracts.js \
    && node output/tools/generateContractInterfaces.js
```
Putting that in the `Dockerfile` will make it slightly easier for future readers to understand what is happening without having to trace through the 6 different scripts that get executed.

If you decide to keep the scripts, recommend making it so the `fetch:sai` script is only executed once on `npm run clean-start`.  At the moment, I believe it will be executed twice (once by `build:contracts` and again by `build:interfaces`).

I also recommend (If you keep NPM scripts) doing `RUN npm run build` rather than `clean-start` since we know we don't neede to clean during docker image generation.

Reply:
Note: If you do decide to clean up the scripts, we may be able to delete some dependencies like `rimraf`.

Reply:
You are right, that it is run twice. The script does check to see nothing needs to be done, without hitting github, but this sounds like a reasonable change.